lib/config/config.go

```diff
@@ -177,29 +182,13 @@ func ReadJSON(r io.Reader, myID protocol.DeviceID) (Configuration, error) {
 	if err := json.Unmarshal(bs, &cfg); err != nil {
 		return Configuration{}, err
 	}
-	cfg.OriginalVersion = cfg.Version
 
 	if err := cfg.prepare(myID); err != nil {
 		return Configuration{}, err
 	}
 	return cfg, nil
 }
 
-type Configuration struct {
-	Version        int                   `xml:"version,attr" json:"version"`
-	Folders        []FolderConfiguration `xml:"folder" json:"folders"`
-	Devices        []DeviceConfiguration `xml:"device" json:"devices"`
-	GUI            GUIConfiguration      `xml:"gui" json:"gui"`
-	LDAP           LDAPConfiguration     `xml:"ldap" json:"ldap"`
-	Options        OptionsConfiguration  `xml:"options" json:"options"`
-	IgnoredDevices []ObservedDevice      `xml:"remoteIgnoredDevice" json:"remoteIgnoredDevices"`
-	PendingDevices []ObservedDevice      `xml:"pendingDevice" json:"pendingDevices"`
-	XMLName        xml.Name              `xml:"configuration" json:"-"`
-
-	MyID            protocol.DeviceID `xml:"-" json:"-"` // Provided by the instantiator.
```

Comment:
> Tell me how much you hate me.

I hate you very much ;-)

Where can I find this piece of information now?

Reply:
Its still on the wrapper I think.

Reply:
Never mind, I found it in `model.id` and agree it does not need to be magically inserted into the configuration struct.